### DISCUSSION TITLE
PERMANOVA effect size for each factor with adonis2

### DISCUSSION THREAD

**Discussion by siobhanschenk** [vegandevs/vegan#709] [Q&A] [answered]
Hi, 
Similarly to issue #704, I'm wondering if it would be possible to show the effect size for each term in a PERMANOVA with adonis2. I can see the individual terms with by="margin" but I would like to see the per-factor effect size without needing to run a marginal PERMANOVA. 

I was able to see individual effect sizes before without by="margin" but not anymore. Not sure if this is R version thing (I'm running 4.4.1 currently) or a change with vegan itself (I'm running 2.6-8).

Thanks!

Current output with marginal PERMANOVA (for context if needed) 

```r
adonis2(formula = dune ~ Management + A1, data = dune.env, permutations = 99, by = "margin")

#            Df SumOfSqs      R2      F Pr(>F)   
# Management  3   1.1865 0.27600 2.4828   0.01 
# A1          1   0.4409 0.10256 2.7676   0.02 
# Residual   15   2.3895 0.55583                 
# Total      19   4.2990 1.00000                
```

**Comment by jarioksa** [comment, marked as answer]
In **vegan** prior to 2.6-8, `adonis2` default was `by = "terms"`. `adonis2` (or `adonis`)  never had a default `by = "margin"` , but the tests were sequential. 

If you want to have statistics `by = "margin"`, you must request tests `by = "margin"`. There is no way of getting marginal effects without running an analysis using marginal effects. There never was.

See the release NEWS for **vegan** 2.6-8:

* `adonis2()` now defaults to running an omnibus test of the model (`by = NULL`) instead of a sequential test of model terms (`by =
  "terms"`). This makes `adonis2()` more consistent with the default for related ordination methods.  See [#677](https://github.com/vegandevs/vegan/issues/677).

**Comment by siobhanschenk** [comment]
Thanks, adding `by = "terms"` gets me the output I wanted. 

Including the output here for anyone reading this later.

```
data(dune)
data(dune.env)
adonis2(dune ~ Management*A1, data=dune.env, permutations=99)
```
_adonis2(formula = dune ~ Management * A1, data = dune.env, permutations = 99)
         Df SumOfSqs      R2      F Pr(>F)   
**Model     7   2.4987 0.58122 2.3792   0.01** 
Residual 12   1.8004 0.41878                 
Total    19   4.2990 1.00000_             


 `adonis2(dune ~ Management*A1, data=dune.env, permutations=99, by="terms")`

_adonis2(formula = dune ~ Management * A1, data = dune.env, permutations = 99, **by = "terms"**)
              Df SumOfSqs      R2      F Pr(>F)   
**Management     3   1.4686 0.34161 3.2629   0.01 
A1             1   0.4409 0.10256 2.9387   0.03 
Management:A1  3   0.5892 0.13705 1.3090   0.21**   
Residual      12   1.8004 0.41878                 
Total         19   4.2990 1.00000_
